tests/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -159,26 +158,40 @@ public function testListDatabases(): void
         self::assertContains('test_create_database', $databases);
     }
 
-    public function testListNamespaceNames(): void
+    /**
+     * @dataProvider listSchemaNamesMethodProvider
+     */
+    public function testListSchemaNames(callable $method): void
     {
         if (! $this->schemaManager->getDatabasePlatform()->supportsSchemas()) {
             self::markTestSkipped('Platform does not support schemas.');
         }
 
-        // Currently dropping schemas is not supported, so we have to workaround here.
-        $namespaces = $this->schemaManager->listNamespaceNames();
-        $namespaces = array_map('strtolower', $namespaces);
+        $this->schemaManager->tryMethod('dropSchema', 'test_create_schema');
 
-        if (! in_array('test_create_schema', $namespaces, true)) {
-            $this->connection->executeStatement(
-                $this->schemaManager->getDatabasePlatform()->getCreateSchemaSQL('test_create_schema')
-            );
+        $this->connection->executeStatement(
+            $this->schemaManager->getDatabasePlatform()->getCreateSchemaSQL('test_create_schema')
+        );
 
-            $namespaces = $this->schemaManager->listNamespaceNames();
-            $namespaces = array_map('strtolower', $namespaces);
-        }
+        self::assertContains('test_create_schema', $method($this->schemaManager));
+    }
 
-        self::assertContains('test_create_schema', $namespaces);
+    /**
+     * @return iterable<list<mixed>>
```

Comment:
dataproviders return signature should match the `return iterable<array{…}>`, where the `…` part is build from the phpdoc of the method that use it, so here `return iterable<array{callable(AbstractSchemaManager): list<string>}>`?

Reply:
> ERROR: InvalidReturnType - tests/Functional/Schema/SchemaManagerFunctionalTestCase.php:180:16 - The declared return type 'iterable<mixed, array{callable(Doctrine\DBAL\Schema\AbstractSchemaManager):list<string>}>' for Doctrine\DBAL\Tests\Functional\Schema\SchemaManagerFunctionalTestCase::listSchemaNamesMethodProvider is incorrect, got 'Generator<int, array{impure-Closure(Doctrine\DBAL\Schema\AbstractSchemaManager):array<array-key, string>|impure-Closure(Doctrine\DBAL\Schema\AbstractSchemaManager):list<string>}, mixed, void>' (see https://psalm.dev/011)
     * @return iterable<array{callable(AbstractSchemaManager): list<string>}>

I agree in general but it would be much better if it was validated by the static analysis. Otherwise, it helps only when implementing the data provider but isn't used by the caller.